acra/security-controls/sql-firewall/_index.md

```diff
@@ -2,3 +2,233 @@
 title: SQL firewall
 bookCollapseSection: true
 ---
+
+## AcraCensor — Acra’s firewall
+
+AcraCensor is a separate SQL firewall module for AcraServer that checks every incoming request. AcraCensor logs SQL requests directed to the database, allows and denies those requests according to certain rules (configured as allowlist/denylist), or ignores them and logs an exception (this might be useful if some requests can’t be parsed by AcraCensor). AcraCensor can be configured for your exact use case, allowing to setup an allowlist and a denylist for tables, query patterns, and exact queries. AcraCensor's goal is to protect the database from SQL injections and suspicious SQL queries.
+
+AcraCensor supports SQL database types — MySQL and PostgreSQL (and their flavours i.e. MariaDB). It is built on top of [xwb1989/sqlparser](https://github.com/xwb1989/sqlparser), which we had extended significantly.
+
+AcraCensor is compatible with SIEM systems, with logs that can be used for alerts' configuration and anomaly detection.
+
+To demonstrate the process of configuring AcraCensor for SQL injection prevention in [OWASP Mutillidae 2 example app](https://github.com/webpwnized/mutillidae), we created [a Docker-based demo project](https://github.com/cossacklabs/acra-censor-demo). Read engineering details about how we built AcraCensor in [our blog](https://www.cossacklabs.com/blog/how-to-build-sql-firewall-acracensor.html).
+
+### Launching AcraCensor
+
+AcraCensor is a built-in component of AcraServer, so it starts running when AcraServer does if the path to its configuration file is provided. You can provide a path for AcraCensor's configuration file using `--acracensor_config_file` parameter in [`acra-server.yaml` configuration](https://github.com/cossacklabs/acra/blob/master/configs/acra-server.yaml).
+
+You can find the basic example configuration file [`configs/acra-censor.example.yaml`](https://github.com/cossacklabs/acra/blob/master/configs/acra-censor.example.yaml) of AcraCensor in our GitHub repository. More real-world examples are available in the [AcraCensor demo project](https://github.com/cossacklabs/acra-censor-demo). You can use these examples as a template for your own, using the database of your choice.
+
+Starting AcraCensor using exact configuration file:
+
+```
+$GOPATH/bin/acra-server --acracensor_config_file=configs/acra-censor.example.yaml --client_id=client_name --incoming_connection_host=acra.connector.host --db_host=postgresql.db.host 
+```
+
+Refer to the [AcraServer]({{< ref "/acra/configuring-maintaining/general-configuration/acra_server.md" >}}) documentation if you have problems with passing configuration parameters.
+
+### Configuring AcraCensor rules
+
+The configuration file describes how AcraCensor should process the incoming SQL queries – log, pass, block, ignore. Each action is described by the corresponding handler. Their position in the file represents their priority of processing.
```

Comment:
IMO, it worth to add phrase here about restarting AcraServer if configuration of AcraCensor is changed.

Reply:
added